app/core/store/modules/unitMap.js

```diff
@@ -0,0 +1,84 @@
+import api from 'core/api'
+
+export default {
+  namespaced: true,
+  state: {
+    currentLevelsList: {}
+  },
+
+  mutations: {
+    setCurrentLevelsList: (state, levels) => {
+      state.currentLevelsList = levels
+    }
+  },
+
+  getters: {
+    getCurrentLevelsList: (state) => state.currentLevelsList
+  },
+
+  actions: {
+    /**
+      We have a campaign.levels list and a classroom.courses.levels list, and they are not always in sync.
+      Hence to get the levels data for a course instance for the unit map, we get the data as follows:
+      1. levels list from the classroom snapshot
+      2. position, nextLevels, first from the classroom snapshot, if does not exist then from campaign snapshot
+      4. any other data from the campaign snapshot, but if doesnt exist in campaign any more then use the data in classroom snapshot
+      */
+    buildLevelsData: async ({ commit, rootGetters, dispatch }, campaignHandle, courseInstanceId) => { // campaignHandle necessary
+      let levels = {}
+      await dispatch('campaigns/fetch', campaignHandle, { root: true })
+      const campaignData = rootGetters['campaigns/getCampaignData'](campaignHandle)
+      if (!courseInstanceId) {
+        levels = campaignData.levels
+      } else {
+        try {
+          const courseInstance = await api.courseInstances.get({ courseInstanceID: courseInstanceId })
```

Comment:
Nitpick: Could this be moved as an addition to the courseInstances store module? Similar to how the campaign data is handled.

Reply:
Yeah potentially, I will add a TODO comment for that here for later in the interest of time.